core/src/main/java/org/mobicents/media/core/configuration/MediaConfiguration.java

```diff
@@ -107,8 +106,8 @@ public void addCodec(String codec) {
         this.codecs.add(codec.toLowerCase());
     }
 
-    public Iterator<String> getCodecs() {
-        return this.codecs.iterator();
+    public String[] getCodecs() {
+        return this.codecs.toArray(new String[]{});
```

Comment:
specify array length (based on codecs.size) to optimize.
http://stackoverflow.com/a/5982478/4142153